src/twisted/internet/test/test_unix.py

```diff
@@ -429,18 +444,91 @@ def _dataReceived(self, data):
 
         dataToSend = b'some data needs to be sent'
         fdsToSend = [sendSocket.fileno(), recvSocket.fileno()]
-        ancillary = [(SOL_SOCKET, SCM_RIGHTS, pack('ii', *fdsToSend))]
+        ancillary, expectedCount = ancillaryPacker(fdsToSend)
         sendmsg(sendSocket, dataToSend, ancillary)
 
         receiver.doRead()
 
         # Verify that fileDescriptorReceived was called twice.
-        self.assertEqual(len(proto.fds), 2)
+        self.assertEqual(len(proto.fds), expectedCount)
 
         # Verify that received FDs are different from the sent ones.
         self.assertFalse(set(fdsToSend).intersection(set(proto.fds)))
```

Comment:
(required)

Hmm.  The existing assertion doesn't verify that the order of file descriptors received is the same as the order of file descriptors sent?  So if the implementation mixed them up we wouldn't notice...  Is that right?

If so, that seems moderately unfortunate.  It's a pre-existing defect so it could be fixed in a follow-up ticket.  Please either fix it here or file that follow-up ticket.